src/btree/row_modify.c

```diff
@@ -243,11 +243,13 @@ __wt_row_modify(WT_CURSOR_BTREE *cbt, const WT_ITEM *key, const WT_ITEM *value,
             __wt_upd_value_assign(cbt->modify_update, upd);
         } else {
             /*
-             * We either insert a tombstone with a standard update or only a standard update to the
+             * If the leaf page of the tree is delta enabled, skip the usual checks for update types
+             * and update chain structure that are required for history store updates. In this case,
```

Comment:
nit: in this case sounds weird. If it refers to the delta enabled case, then the following words doesn't make sense. Should be otherwise?

Reply:
@quchenhao Does this look good ?
```
            /*
             * If this refers to the delta-enabled case, the following checks do not make sense. We
             * either insert a tombstone with a standard update or only a standard update to the
             * history store if we write a prepared update to the data store.
             */
```

Reply:
`If this refers to the delta-enabled case, the following checks do not make sense` may be we say we can skip the check?

Reply:
Fixed in 7ee35faf65a9be8dbe6a8544706e700093736920